### PR TITLE
Adds nightly doc building to travis

### PR DESCRIPTION
**Pull request by dkliban**
The 2-master branch will be setup to build using cron on Travis. When this build is trigerred, the
docs for branches defined in .travis/2-master.yaml will be built and published.

re: #3722
https://pulp.plan.io/issues/3722

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,27 @@
 language: python
 python:
-    - "2.7"
+- '3.6'
```
Comment (daviddavis) [inline .travis.yml]: Minor grip but typically arrays in yaml are indented.

File: .travis/docs/all_content_index.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Documentation Index
```
Comment (daviddavis) [inline .travis/docs/all_content_index.rst]: I feel like these files aren't necessarily specific to Travis? Thoughts on maybe moving these to the top level docs directory?

File: .travis.yml (hunk 1 above)
Comment (daviddavis) [reply .travis.yml]: In the files below (e.g. `.travis/2-master.yaml`) you're indenting array entries.

File: .travis/docs/all_content_index.rst (hunk 2 above)
Comment (dkliban) [reply .travis/docs/all_content_index.rst]: They are specific to the way we build docs for docs.pulpproject.org. You can build the docs in this repo by themselves and they will not include plugin information.

**Pull request merged**